libs-scala/gatling-utils/src/main/scala/com/daml/gatling/stats/ConvertStats.scala

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.gatling
+package stats
+
+import java.io.File
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+
+import com.daml.gatling.stats.ConvertStats.StatsKey
+import com.daml.scalautil.Statement.discard
+import scalaz.\/
+
+/*
+ * Provides the ability to parse from Gatling's JSON format, given a set of relevant stats
+ * to look for and a way of handling them.
+ */
+trait ConvertStats[StatsT] {
+  def relevantStats: List[StatsKey]
+
+  def fromSimulationLog(simulation: SimulationLog): String \/ StatsT
+
+  def parse(log: SimulationLog): String \/ StatsT = fromSimulationLog(log)
+}
+
+object ConvertStats {
+  type StatsKey = String
+  def apply[S](implicit R: ConvertStats[S]): ConvertStats[S] = R
+}
+
+object SimulationLogConvertSyntax extends SimulationLogConvertSyntax
+
+trait SimulationLogConvertSyntax {
+  implicit class SimulationLogConvertOps(val log: SimulationLog) {
+    def parseStats[S: ConvertStats] = ConvertStats[S].parse(log)
+    def parseFormattableStats[S](implicit S: ConvertWithFallback[Formattable, S]) = S.convert(log)
+    def readAnyStats: String \/ Formattable =
+      convertFormattableStats[CombinedStats || FixTradeStats || SchedulerStats]
```

Comment:
Might be better to combine this change with implementing two of your own test result statistics classes. I don't think you are dealing with a scheduler (single request => two async responses), nor with the FIX protocol.
I understand having an example of how to implement the combined reader in the first version is worthwhile but better done using your own use case.

Reply:
Actually I don't need to split stats. I just print the unfiltered stats from the simulation log. And you are correct I don't need FixTradeStats and SchedulerStats. Thanks @jberthold-da!